packages/ember-headless-form/rollup.config.mjs

```diff
@@ -16,9 +16,12 @@ export default {
     // These are the modules that users should be able to import from your
     // addon. Anything not listed here may get optimized away.
     addon.publicEntrypoints([
-      'components/**/*.js',
```

Comment:
Yeah, makes sense!

I applied your suggestion (`**/*.js` as rollup entry points) and restricted the package exports [here](https://github.com/CrowdStrike/ember-headless-form/pull/42/files#diff-58527bca9ef9ea46409bde3010f934f388d9db44ff565826a485209702cd4ad3R109). Note that I did not say that only our index module is our public module, because we still need `/components/headless-form` to be importable due to the app-reexport mapping [here](https://github.com/CrowdStrike/ember-headless-form/pull/42/files#diff-58527bca9ef9ea46409bde3010f934f388d9db44ff565826a485209702cd4ad3R100) (I don't think we can use a _named export_ from index instead). Instead I am flagging `-private/*` as not exported (TIL about `null` being a valid value here, see https://nodejs.org/api/packages.html#package-entry-points). Confirmed that it also works for webpack (i.e. causes build error when trying to import).

> One reason being is that it's not on us to optimize our assets -- we emit dev-time assets, and it's the responsibility of the consuming app to decide how to bundle.

Yeah, although I don't think `rollup-plugin-ts` considers itself to be a plugin specifically for _app_ builds, with its type declarations support it targets builders of libraries like us. Yet it tries to be smart about tree-shaking declarations and whatnot. I guess this wouldn't be bad (e.g. prevent internal types to accidentally be leaked to the public) - if it worked reliably (which it doesn't)! But I also agree, we can handle that as well or better on our own with package exports!